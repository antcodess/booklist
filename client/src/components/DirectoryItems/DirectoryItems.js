import React from 'react';
import Slider from 'react-slick';
import { withRouter } from 'react-router-dom';

import DirectoryItem from '../DirectoryItem/DirectoryItem';

import 'slick-carousel/slick/slick.css';
import 'slick-carousel/slick/slick-theme.css';
import './DirectoryItems.scss';


const DirectoryItems = ({ title, items, history, match }) => {
  
  const settings = {
    dots: true,
    infinite: items.length > 3 ? true : false,
    speed: 500,
    slidesToShow: 4,
    slidesToScroll: 4,
    swipeToSlide: true,
    arrows: false,
    responsive: [
      {
        breakpoint: 940,
        settings: {
          slidesToShow: 3,
          slidesToScroll: 3,
        }
      },
      {
        breakpoint: 700,
        settings: {
          slidesToShow: 2,
          slidesToScroll: 2,
        }
      },
      {
        breakpoint: 600,
        settings: {
          slidesToShow: 1,
          slidesToScroll: 1
        }
      }
    ]
  }; 

  const categoryURL = encodeURIComponent(title.replace(/[\s]/g, '-').toLowerCase());

  return (
    <div className='directory-items-container'>
      <div className='directory-items-title'>
        <h3 onClick={() => history.push(`/categories/${categoryURL}`)}>
          { title }
        </h3>
      </div>
      <div className='directory-items'>
        <Slider { ...settings }>
          {items
            .filter((item, index) => index < 6)
            .map((item) => (
              <DirectoryItem key={item.id} item={item} />
            ))
          }
        </Slider>
      </div>
    </div>
  );
};

export default withRouter(DirectoryItems);