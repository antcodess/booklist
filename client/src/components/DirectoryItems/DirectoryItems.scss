.directory-items-container {
  margin-bottom: 60px; 

  .directory-items-title {
    display: flex;

    .directory-items-title-link {
      align-items: center;
      font-family: Helvetica, Arial, "Lucida Grande", sans-serif;
      font-weight: bolder;
      font-size: 0.875rem;
      margin-right: 5px;

      &:hover {
        color: #0080DE;
      }
  
    }
    h3 {
      margin-bottom: 30px;
      cursor: pointer;

      &:hover {
        opacity: 0.7;
      }
    }
  }

  .directory-items {
    vertical-align: baseline;

    .slick-prev:before, .slick-next:before {
      // color: black;
      font-size:40px;
      opacity: 1;
    }

    .slick-slide {
      padding: 0 5px;
    }

    .slick-dots li button:before {
      font-size: 13px;
      line-height: 40px;
    }

    .slick-dots li.slick-active button:before {
      opacity: 1;
    }

    .slick-prev {
      left: -42px;
    }
  }
}

@media only screen and (max-width: 1500px) {
  .directory-items-container {
    .directory-items {
      .slick-prev, .slick-next {
        top: 125px;
      }
    }
  } 
}

@media only screen and (max-width: 1100px) {
  .directory-items-container {
    .directory-items {
      .slick-prev, .slick-next {
        top: 95px;
      }
    }
  }
}

@media only screen and (max-width: 600px) {
  .directory-items-container {
    h3 {
      justify-content: center;
    }
    .directory-items-title {
      justify-content: center;
    }
    .directory-items {
      .slick-prev, .slick-next{
        top: 155px;
      }
    }
  } 
}

@media only screen and (max-width: 500px) {
  .directory-items-container {
    .directory-items {
      .slick-prev, .slick-next {
        top: 130px;
      }
    }
  }
}

@media only screen and (max-width: 400px) {
  .directory-items-container {
    .directory-items {
      .slick-prev, .slick-next {
        top: 83px;
      }
    }
  } 
}