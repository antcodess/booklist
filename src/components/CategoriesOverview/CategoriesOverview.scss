.category-overview {
  margin-top: 20px;
}