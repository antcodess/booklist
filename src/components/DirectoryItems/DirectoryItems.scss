.directory-items-container {
  margin-bottom: 60px; 

  h3 {
    display: flex;
    margin-bottom: 30px;
  }

  .directory-items {
    // width: 100%;
    // display: flex;
    // justify-content: space-between;
    // flex-wrap: wrap;
    // .slick-prev {
    //   left: -32px;
    // }
    
    .slick-prev:before, .slick-next:before {
      color: black;
      // font-size: 28px;
      opacity: 1;
    }

    .slick-slide {
      padding: 0 5px;
    }

    // .slick-prev, .slick-next {
    //   top: 95px;
    // }

    .slick-dots li button:before {
      font-size: 13px;
      line-height: 40px;
    }

    .slick-dots li.slick-active button:before {
      opacity: 1;
    }
  }
}