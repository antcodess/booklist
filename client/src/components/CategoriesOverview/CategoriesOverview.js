import React from 'react';
import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';
import { selectAllProductCategories } from '../../redux/shop/shop.selectors';

import DirectoryItems from '../DirectoryItems/DirectoryItems';

import './CategoriesOverview.scss';

export const CategoryOverview = ({ categories }) => {
  return (
    <section className='category-overview'>
      {
        categories.map((category, index) => 
        <DirectoryItems key={index} 
                        title={category.categoryName} 
                        items={category.items}  />)
      }
    </section>
  );
};

const mapStateToProps = createStructuredSelector({
  categories: selectAllProductCategories
});

export default connect(mapStateToProps)(CategoryOverview);