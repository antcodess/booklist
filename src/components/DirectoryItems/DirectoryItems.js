import React from 'react';
import Slider from 'react-slick';
import { Link } from 'react-router-dom';
import DirectoryItem from '../DirectoryItem/DirectoryItem';
import 'slick-carousel/slick/slick.css';
import 'slick-carousel/slick/slick-theme.css';

import './DirectoryItems.scss';


const DirectoryItems = ({ title, items }) => {
  const settings = {
    dots: true,
    infinite: true,
    speed: 500,
    slidesToShow: 4,
    slidesToScroll: 4,
    swipeToSlide: true,
    arrows: false,
    responsive: [
      {
        breakpoint: 900,
        settings: {
          slidesToShow: 3,
          slidesToScroll: 3,
        }
      },
      {
        breakpoint: 700,
        settings: {
          slidesToShow: 2,
          slidesToScroll: 2,
        }
      },
      {
        breakpoint: 600,
        settings: {
          slidesToShow: 1,
          slidesToScroll: 1
        }
      }
    ]
  };

  const categoryURL = title.replace(/[\s]/g, '-').toLowerCase();

  return (
    <div className='directory-items-container'>
      <div className='directory-items-title'>
        <h3>{ title }</h3>
        <Link className='directory-items-title-link' to={`categories/${categoryURL}`}>View All</Link>
      </div>
      <div className='directory-items'>
        <Slider { ...settings }>
          {items
            .filter((item, index) => index < 6)
            .map((item) => (
              <DirectoryItem key={item.id} item={item} />
            ))
          }
        </Slider>
      </div>
    </div>
  );
};

export default DirectoryItems;