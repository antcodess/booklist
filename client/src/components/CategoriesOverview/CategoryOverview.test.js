import React from 'react';
import { shallow } from 'enzyme';
import { CategoryOverview } from './CategoriesOverview';
import DirectoryItems from '../DirectoryItems/DirectoryItems';

describe('CategoryOverview Component', () => {
  let wrapper;
  let mockCategories;
  
  beforeEach(() => {

    mockCategories = [
      {
        categoryName: 'Category 1',
        items: [{id: 1}, {id: 2}, {id:3}]
      },
      {
        categoryName: 'Category 2',
        items: [{id: 1}, {id: 2}, {id:3}]
      },
      {
        categoryName: 'Category 3',
        items: [{id: 1}, {id: 2}, {id:3}]
      }
    ];

    const mockProps = {
      categories: mockCategories
    };

    wrapper = shallow(<CategoryOverview { ...mockProps } />);
  });

  it('should render CategoryOverview Component', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render correct amount of DirectoryItems as categories length', () => {
    expect(wrapper.find(DirectoryItems).length).toEqual(mockCategories.length)
  });
});