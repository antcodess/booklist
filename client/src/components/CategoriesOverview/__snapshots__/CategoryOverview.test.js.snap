// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoryOverview Component should render CategoryOverview Component 1`] = `ShallowWrapper {}`;
